octavia_proxy/tests/unit/api/controllers/test_base.py

```diff
@@ -0,0 +1,270 @@
+#    under the License.
+from unittest import mock
+
+from oslo_config import cfg
+from otcextensions.sdk.vlb.v3 import (
+    load_balancer as elbv3,
+    listener as lsv3,
+    pool as plv3,
+    member as memv3,
+    health_monitor as hmv3,
+    l7_policy as l7pv3,
+    l7_rule as l7rv3,
+    flavor as flv3
+)
+
+from octavia_proxy.api.v2.controllers import base as _b
+from octavia_proxy.common import exceptions
+from octavia_proxy.tests.unit.api.controllers import base
+
+CONF = cfg.CONF
+
+
+class TestBaseController(base.TestCase):
+    def setUp(self):
+        super().setUp()
+        CONF.api_settings.enabled_provider_drivers = {
+            'elbv3': 'The ELBv3 driver.',
+        }
+
+        self.bc = _b.BaseController
+
+        self.context = mock.MagicMock()
+        self.context.session = mock.MagicMock()
+        self.context.project_id = 'id'
+        self.session = self.context.session
+
+    def test_basic(self):
+        self.assertEqual(self.bc.RBAC_TYPE, None)
+        self.assertIsNotNone(getattr(self.bc, 'find_load_balancer'))
+        self.assertIsNotNone(getattr(self.bc, 'find_listener'))
+        self.assertIsNotNone(getattr(self.bc, 'find_pool'))
+        self.assertIsNotNone(getattr(self.bc, 'find_member'))
+        self.assertIsNotNone(getattr(self.bc, 'find_health_monitor'))
+        self.assertIsNotNone(getattr(self.bc, 'find_l7policy'))
+        self.assertIsNotNone(getattr(self.bc, 'find_l7rule'))
+        self.assertIsNotNone(getattr(self.bc, 'find_flavor'))
+        self.assertIsNotNone(getattr(self.bc, '_convert_sdk_to_type'))
+        self.assertIsNotNone(getattr(self.bc, '_validate_protocol'))
+        self.assertIsNotNone(
+            getattr(self.bc, '_is_only_specified_in_request')
+        )
+        self.assertIsNotNone(
+            getattr(self.bc, '_validate_pool_request_for_tcp_udp')
+        )
+        self.assertIsNotNone(
+            getattr(self.bc, '_validate_healthmonitor_request_for_udp')
+        )
+        self.assertIsNotNone(getattr(self.bc, '_auth_get_all'))
+        self.assertIsNotNone(getattr(self.bc, '_auth_validate_action'))
+        self.assertIsNotNone(getattr(self.bc, '_filter_fields'))
+        self.assertIsNotNone(getattr(self.bc, '_get_attrs'))
+
+    def test_find_load_balancer(self):
+        lb_v3 = elbv3.LoadBalancer(
+            id='07f0a424-cdb9-4584-b9c0-6a38fbacdc3a',
+            name='test_v3',
+            created_at='2021-08-10T09:39:24+00:00',
+            updated_at='2021-08-10T09:39:24+00:00'
+        )
+        self.session.vlb.find_load_balancer = mock.MagicMock(
+            return_value=lb_v3
+        )
+        lb = self.bc().find_load_balancer(self.context, 'id')
+
+        self.assertIsNotNone(lb)
+        self.session.vlb.find_load_balancer.assert_called_with(
+            ignore_missing=True,
+            name_or_id='id'
+        )
+
+    def test_find_load_balancer_not_found(self):
```

Comment:
Maybe it might be better to use test_load_balancer_not_found ?

Reply:
here i try to test exact find_*** functions, because of that i prefer to keep find word in test name